site/public/js/gradeable.js

```diff
@@ -373,6 +373,30 @@ function ajaxUploadGradeable() {
     return true;
 }
 
+// Download the JSON file to recreate the gradeable.
+function downloadGradeableJson($gradeable_id) {
+    const url = buildCourseUrl([$gradeable_id, 'download']);
+    fetch(url, {
+        method: 'GET',
+    }).then(
+        response => {
+            return response.json();
+        }).then(json => {
+        if (json['status'] === 'success') {
+            const temporaryElement = document.createElement('a');
+            const file = new Blob([JSON.stringify(json['data'], null, 4)]);
+            temporaryElement.href = URL.createObjectURL(file);
+            temporaryElement.download = `${json['data']['id']}.json`;
+            temporaryElement.click();
+        }
+        else {
+            displayErrorMessage(json['message']);
+            return false;
+        }
+    });
+    return true;
+}
```

Comment:
Why is this necessary?  Can't you just send it with the appropriate headers set on the server side such that it gets downloaded automatically?

Reply:
I am not quite sure what you mean by that, could you explain a bit more?